### PR TITLE
New backend delete

### PR DESCRIPTION
**Pull request by nebril**
Extracted delete specific code from ReleaseServer to external function
which is called from both Local and Remote ReleaseModules.

Made getVersionSet function from tiller package exported.

### CODE REVIEW

File: cmd/rudder/rudder.go [hunk 1]
```diff
@@ -69,13 +79,37 @@ func (r *ReleaseModuleServiceServer) InstallRelease(ctx context.Context, in *rud
...
+// DeleteRelease deletes a provided release
 func (r *ReleaseModuleServiceServer) DeleteRelease(ctx context.Context, in *rudderAPI.DeleteReleaseRequest) (*rudderAPI.DeleteReleaseResponse, error) {
 	grpclog.Print("delete")
-	return nil, nil
+
+	resp := &rudderAPI.DeleteReleaseResponse{}
+	rel := in.Release
+	vs, err := tiller.GetVersionSet(clientset.Discovery())
+	if err != nil {
+		return resp, fmt.Errorf("Could not get apiVersions from Kubernetes: %s", err)
```
Comment (jellonek) [inline cmd/rudder/rudder.go]: Looks like in other places there is used `%v` for err formatting.

File: pkg/tiller/release_modules.go [hunk 2]
```diff
@@ -59,6 +70,15 @@ func (m *LocalReleaseModule) Status(r *release.Release, req *services.GetRelease
 	return env.KubeClient.Get(r.Namespace, bytes.NewBufferString(r.Manifest))
 }
 
+// Delete deletes the release and returns manifests that were kept in the deletion process
+func (m *LocalReleaseModule) Delete(rel *release.Release, req *services.UninstallReleaseRequest, env *environment.Environment) (kept string, errs []error) {
+	vs, err := GetVersionSet(m.clientset.Discovery())
+	if err != nil {
+		return rel.Manifest, []error{fmt.Errorf("Could not get apiVersions from Kubernetes: %s", err)}
```
Comment (jellonek) [inline pkg/tiller/release_modules.go]: Same again

File: pkg/tiller/release_modules.go [hunk 3]
```diff
@@ -100,3 +120,44 @@ func (m *RemoteReleaseModule) Status(r *release.Release, req *services.GetReleas
 	resp, err := rudder.ReleaseStatus(statusRequest)
 	return resp.Info.Status.Resources, err
 }
+
+func (m *RemoteReleaseModule) Delete(r *release.Release, req *services.UninstallReleaseRequest, env *environment.Environment) (string, []error) {
```
Comment (jellonek) [inline pkg/tiller/release_modules.go]: Missing docstring.

**Review by jellonek — COMMENTED** [review]
Minor things, rest lgtm.

File: cmd/rudder/rudder.go (hunk 1 above)
Comment (nebril) [reply cmd/rudder/rudder.go]: done

File: pkg/tiller/release_modules.go (hunk 2 above)
Comment (nebril) [reply pkg/tiller/release_modules.go]: done

File: pkg/tiller/release_modules.go (hunk 3 above)
Comment (nebril) [reply pkg/tiller/release_modules.go]: done

File: cmd/rudder/rudder.go [hunk 4]
```diff
@@ -24,16 +24,26 @@ import (
...
-var kubeClient = kube.New(nil)
+var kubeClient *kube.Client
+var clientset internalclientset.Interface
 
 func main() {
+	var err error
+	kubeClient = kube.New(nil)
+	clientset, err = kubeClient.ClientSet()
+	if err != nil {
+		grpclog.Fatalf("Cannot initialize Kubernetes connection: %s", err)
```
Comment (jellonek) [inline cmd/rudder/rudder.go:44]: Ups. Same there ;)

**Pull request merged**